### PR TITLE
add Netty sanity check tests

### PR DESCRIPTION
**Pull request by sullis**
### Type of change

- Enhancement / new feature

### Description

- verify io_uring transport
- verify epoll transport

### Additional Context

Validate presence of Netty transports.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/NettyTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public void epollIsAvailableOnLinux() {
+        assertTrue(Epoll.isAvailable());
+    }
+
+    @Test
+    @EnabledOnOs(value = OS.LINUX)
+    public void ioUringIsAvailableOnLinux() {
+        assertTrue(IOUring.isAvailable());
+    }
+}
```
Comment (k-wall) [inline kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/NettyTest.java:30]: As the project ships kqueue for the Mac/FreeBSD it seems odd to to asset that true.

File: kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/NettyTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import io.netty.channel.epoll.Epoll;
+import io.netty.incubator.channel.uring.IOUring;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class NettyTest {
+    @Test
+    @EnabledOnOs(value = OS.LINUX)
+    public void epollIsAvailableOnLinux() {
+        assertTrue(Epoll.isAvailable());
```
Comment (k-wall) [inline kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/NettyTest.java:22]: the project's favouring assert-j in new tests.

File: kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/NettyTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+
+import io.netty.channel.epoll.Epoll;
+import io.netty.incubator.channel.uring.IOUring;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class NettyTest {
```
Comment (k-wall) [inline kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/NettyTest.java:18]: NettyTransportTest

**Comment by SamBarker** [comment]
Do you have a particular need for these tests? To me they are just validating that we build with the dependencies we declare. It might be much more interesting if we can run this as part of image building or something to assert that the container had the desired libraries but I'm not sure what value they add at the junit stage.

**Comment by SamBarker** [comment]
@sullis can you shed some light on your motivation for this PR?

**Comment by SamBarker** [comment]
@sullis Thanks for your interest in Kroxylicious and we are always interested in improving the value of our tests suite however we aren't convinced that these tests do anything more than validate that maven built the requested test classpath so see little value in them. 

Please reopen the PR and show us what is wrong with our understanding if you disagree.

**Pull request closed without merging**